src/watts/plugin_pyarc.py

```diff
@@ -0,0 +1,186 @@
+        results_data["rzmflx"] = user_object.results_keff_rzmflx
+        results_data["persent_pert"] = user_object.results_kpert_persent
+        results_data["persent_sens"] = user_object.results_kuq_persent
+        results_data["gamsor"] = user_object.results_power_gamsor
+        results_data["dassh"] = user_object.results_dassh
+        # TODO - this is minimum information that can be brought back easily. 
+        # More should be returned but will require work on PyARC.
+        return results_data
+
+    def save(self, filename: PathLike):
+        """Save results to an HDF5 file
+
+        Parameters
+        ----------
+        filename
+            File to save results to
+        """
+        with h5py.File(filename, 'w') as h5file:
+            super()._save(h5file)
+
+    @classmethod
+    def _from_hdf5(cls, obj: h5py.Group):
+        """Load results from an HDF5 file
+
+        Parameters
+        ----------
+        obj
+            HDF5 group to load results from
+        """
+        time, parameters, inputs, outputs = Results._load(obj)
+        return cls(parameters, time, inputs, outputs)
+
+class PluginPyARC(TemplatePlugin):
+    """Plugin for running PyARC
+
+    Parameters
+    ----------
+    template_file
+        Templated PyARC input
+    show_stdout
+        Whether to display output from stdout when SAM is run
+    show_stderr
+        Whether to display output from stderr when SAM is run
+    supp_inputs
+        List of supplementary input files that are needed for running PyARC
+    
+    Attributes
+    ----------
+    pyarc_exec
+        Path to PyARC executable
+
+    """
+
+    def  __init__(self, template_file: str, show_stdout: bool = False,
+                  show_stderr: bool = False, supp_inputs: List[str] = []):
+        super().__init__(template_file)
+        self._pyarc_exec = Path('PyARC.py')
+#        if os.environ["PyARC_DIR"] is not None: # how would I setup this as default exec if not specified?
+#            self._pyarc_exec = os.environ["PyARC_DIR"]
+        self.pyarc_inp_name = "pyarc_input.son"
+        self.show_stdout = show_stdout
+        self.show_stderr = show_stderr
+        self.supp_inputs = [Path(f).resolve() for f in supp_inputs]
+
+    @property
+    def pyarc_exec(self) -> Path:
+        return self._pyarc_exec
+
+    @pyarc_exec.setter
+    def pyarc_exec(self, exe: PathLike):
+        if os.path.exists(exe) is False:
+            raise RuntimeError(f"PyARC executable '{exe}' is missing.")
+        self._pyarc_exec = Path(exe)
+        
+
+    def prerun(self, params: Parameters) -> None:
+        """Generate PyARC input files
+
+        Parameters
+        ----------
+        params
+            Parameters used by the PyARC template
+        """
+        print("Pre-run for PyARC Plugin")
+        self._run_time = time.time_ns()
+        super().prerun(params, filename=self.pyarc_inp_name)
+        sys.path.insert(0, '{}'.format(self._pyarc_exec))
```

Comment:
I don't think this should be the responsibility of watts. If someone wants to use PyARC (which requires them to `import PyARC`), shouldn't we just assume they already have it setup correctly?

Reply:
resolved - we restored sys.path after PyARC execution.